dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java

```diff
@@ -0,0 +1,269 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.kernel.service.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.ffdc.FFDCFilter;
+
+/**
+ * API for getting cpu info about the system
+ */
+public class CpuInfo {
+
+    /**
+     * Trace component for this class.
+     */
+    private final static TraceComponent tc = Tr.register(CpuInfo.class);
+
+    private static CpuInfo instance;
+    private final int AVAILABLE_PROCESSORS;
+
+    // For CPU usage calculation
+    private final OperatingSystemMXBean osmx;
+    private final MBeanServer mBeanServer;
+    private ObjectName operatingSystemMbean;
+    final private int cpuNSFactor;
+    private long lastProcessCPUTime = 0;
+    private double lastProcessCpuUsage = -1;
+    private long lastSystemTimeMillis = -1;
+
+    private CpuInfo() {
+        // find available processors
+        int runtimeAvailableProcessors = Runtime.getRuntime().availableProcessors();
+        int fileSystemAvailableProcessors = getAvailableProcessorsFromFilesystem();
+
+        if (fileSystemAvailableProcessors <= 0 || fileSystemAvailableProcessors > runtimeAvailableProcessors) {
+            AVAILABLE_PROCESSORS = runtimeAvailableProcessors;
+        } else {
+            AVAILABLE_PROCESSORS = fileSystemAvailableProcessors;
+        }
+
+        // initialize mbean required to get CPU usage info
+        osmx = ManagementFactory.getOperatingSystemMXBean();
+        mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+        try {
+            operatingSystemMbean = new ObjectName("java.lang", "type", "OperatingSystem");
+        } catch (MalformedObjectNameException e) {
+            FFDCFilter.processException(e, getClass().getName(), "CpuInfo<init>");
+        }
+
+        int nsFactor = 1;
+        // adjust for J9 cpuUsage units change from hundred-nanoseconds to nanoseconds in Java8sr5
```

Comment:
Might be good to check OpenJ9 here as well. Since the first OpenJ9 release already had the change that you are checking for starting in IBM JDK SR5 it may be sufficient to just check for the vendor being "openj9".

Reply:
@vijaysun-omr Why wouldn't openj9 just do nsFactor =1 like what it is doing with the code as written?  openj9 != ibm from JavaInfo.  OpenJDK, Oracle, and OpenJ9 all should behave the same as expected with the code as written right?

Reply:
Sorry I was looking at the trace message that seems to print a message if it's IBM JDK but won't print it if it's an OpenJ9 build. But maybe that's okay since there is no ambiguity for OpenJ9 (as it released with the >= Java8 SR5 behaviour).